cmake/pcl_all_in_one_installer.cmake

```diff
@@ -30,6 +29,19 @@ foreach(dep Eigen Boost Qhull FLANN VTK)
   list(APPEND PCL_3RDPARTY_COMPONENTS ${dep})
 endforeach()
 
+set(dep_Eigen3 Eigen3)
+install(
+  DIRECTORY "${Eigen3_DIR}/"
+  DESTINATION 3rdParty/${dep_Eigen3}/share/eigen3/cmake
+  COMPONENT ${dep_Eigen3}
+)
+install(
+  DIRECTORY "${Eigen3_INCLUDE_DIR}/"
+  DESTINATION 3rdParty/${dep_Eigen3}/include/eigen3
+  COMPONENT ${dep_Eigen3}
+)
+list(APPEND PCL_3RDPARTY_COMPONENTS ${dep_Eigen3})
```

Comment:
Not sure about this. It's not tested, yet, but it should work. However, it relies on Eigen internals: The cmake files in `${Eigen3_DIR}` expect the include directory to be located at `${Eigen3_DIR}/../../../include/eigen3`, but this is defined by Eigen and might change without notice.

Actually the benefit of using Eigens native cmake integration would be to avoid dependencies to Eigen internals and to rely on their `Eigen3Config.cmake` instead. Thus, adding such things doesn't feel right..

Reply:
Tried to remove the dependencies to Eigen3 internals as much as possible with [718a0cd](https://github.com/PointCloudLibrary/pcl/pull/5613/commits/718a0cda7e503aadfa6dbdc2fc3660e3185c47d7).

However, if Eigen3 would stop to export `EIGEN3_INCLUDE_DIR` as well as `EIGEN3_INCLUDE_DIRS` and would change the include directory from `${Eigen3_DIR}/../../../include/eigen3/` to something else, this approach would still fail.